tests/unit/test_repos_release.py

```diff
@@ -140,6 +141,25 @@ def test_download(self):
         )
         assert stream.called is False
 
+    def test_download_with_302(self):
+        """Verify the request to download an Asset file."""
+        with mock.patch.object(github3.models.GitHubCore, '_get') as get:
```

Comment:
@sigmavirus24 and I are discussing how to properly test this without mocking itself.